include/bitprim/mining/mempool_v1.hpp

```diff
@@ -0,0 +1,1909 @@
+        //     for (auto i : out_removed) {
+        //         std::cout << i << ", ";
+        //     }
+        //     std::cout << std::endl;
+        // }
+
+        // if (node.children_size() != size) {
+        //     std::cout << "node_index:           " << node_index << std::endl;
+        //     std::cout << "node.children_size(): " << node.children_size() << std::endl;
+        //     std::cout << "size:                 " << size << std::endl;
+
+        //     std::cout << "Removed:  ";
+        //     for (auto i : out_removed) {
+        //         std::cout << i << ", ";
+        //     }
+        //     std::cout << std::endl;
+        // }
+
+        // if (node.children_sigops() != sigops) {
+        //     std::cout << "node_index:             " << node_index << std::endl;
+        //     std::cout << "node.children_sigops(): " << node.children_sigops() << std::endl;
+        //     std::cout << "sigops:                 " << sigops << std::endl;
+
+
+        //     std::cout << "Removed:  ";
+        //     for (auto i : out_removed) {
+        //         std::cout << i << ", ";
+        //     }
+        //     std::cout << std::endl;
+        // }
+
+        BITCOIN_ASSERT(node.children_fees() == fee);
+        BITCOIN_ASSERT(node.children_size() == size);
+        BITCOIN_ASSERT(node.children_sigops() == sigops);
+    }
+
+
+    void check_invariant() const {
+        check_invariant_partial();
+
+        {
+            size_t i = 0;
+            for (auto const& node : all_transactions_) {
+                if (node.candidate_index() != null_index) {
+                    for (auto pi : node.parents()) {
+                        auto const& parent = all_transactions_[pi];
+                        BOOST_ASSERT(parent.candidate_index() != null_index);
+                    }
+                }
+                ++i;
+            }
+        } 
+
+        {
+            // size_t ci = 0;
+            for (auto i : candidate_transactions_) {
+                auto const& node = all_transactions_[i];
+                // BOOST_ASSERT(ci == node.candidate_index());
+                // ++ci;
+                check_children_accum(i);
+            }
+        }
+
+        {
+            size_t i = 0;
+            for (auto const& node : all_transactions_) {
+                check_children_accum(i);
+                ++i;
+            }
+        }        
+
+
+        {
+            auto const cmp = [this](index_t a, index_t b) {
+                return fee_per_size_cmp(a, b);
+            };
+
+            auto res = std::is_sorted(candidate_transactions_.begin(), candidate_transactions_.end(), cmp);
+
+            // if (! res) {
+                // auto res2 = std::is_sorted(candidate_transactions_.begin(), candidate_transactions_.end(), cmp);
+                // std::cout << res2;
+            // }
+
+            BOOST_ASSERT(res);
+        }        
+    }
+
+    void check_invariant_partial() const {
+        
```

Comment:
This method is too long, wouldn't it be more readable breaking it into inlined submethods?

Reply:
Marked with a TODO